hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/snappy/SnappyDecompressor.java

```diff
@@ -276,13 +258,29 @@ public void end() {
     // do nothing
   }
 
-  private native static void initIDs();
+  private int decompressBytesDirect() throws IOException {
+    if (compressedDirectBufLen == 0) {
+      return 0;
+    } else {
+      // Set the position and limit of `compressedDirectBuf` for reading
+      compressedDirectBuf.limit(compressedDirectBufLen).position(0);
+      // There is compressed input, decompress it now.
+      int size = Snappy.uncompressedLength((ByteBuffer) compressedDirectBuf);
+      if (size > uncompressedDirectBuf.remaining()) {
+        throw new IOException("Could not decompress data. " +
+          "uncompressedDirectBuf length is too small.");
```

Comment:
use name of config option which users can tun

Reply:
I found this check is not needed. Removed.